drizzlepac/helpers/utils.py

```diff
@@ -0,0 +1,245 @@
+
+    kwargs: dict
+        Additional arguments to be passed to `FITSDiff`
+
+    Returns
+    -------
+    None
+        Assert left and right are identical
+    """
+    assert isinstance(left, str)
+    assert isinstance(right, str)
+
+    local_kwargs = dict(
+        keys=kwargs.get('keys', default_compare['keys']),
+        rtol=kwargs.get('rtol', default_compare['rtol']),
+        ignore_keywords = kwargs.get('ignore_keywords',
+                                     default_compare['ignore_keywords'])
+    )
+
+    keys = local_kwargs['keys']
+    rtol = local_kwargs['rtol']
+    ignore_keywords = local_kwargs['ignore_keywords']
+
+    assert isinstance(keys, list)
+    assert isinstance(rtol, float)
+    assert isinstance(ignore_keywords, list)
+
+    with fits.open(left) as a:
+        with fits.open(right) as b:
+            result = fits.diff.FITSDiff(fits.HDUList([a[kw] for kw in keys]),
+                                        fits.HDUList([b[kw] for kw in keys]),
+                                        ignore_keywords=ignore_keywords,
+                                        rtol=rtol,
+                                        **kwargs)
+
+    if no_assert:
+        return result
+
+    assert result.identical, result.report()
+
+
+def cmp_gen_hdrkeywords(fitsobj, base, keywords, limit=0, start=0):
+    """Generate list of FITS header elements to compare
+
+    TODO: Use correct terminology
+
+    Parameters
+    ----------
+    fitsobj: HDUList
+        TODO
+
+    base: str
+        Primary keyword
+
+    keywords: list
+        Additional keywords to use
+
+    limit: int
+        Number of extensions
+        Note: 1-indexed
+
+    start: int
+        Start with extension number
+
+    Returns
+    -------
+    output: list
+        Keywords to compare
+
+    """
+    assert isinstance(fitsobj, fits.HDUList)
+    assert isinstance(base, str)
+    assert isinstance(keywords, list)
+
+    output = list(fitsobj[base])
+
+    if limit and not start:
+        start += 1
+
+    for idx in range(start, limit + 1):
+        for key in keywords:
+            if not limit:
+                output.append(fitsobj[key])
+            else:
+                output.append(fitsobj[key, idx])
+
+    return output
+
+
+# Check strings based on words using length precision
+def word_precision_check(str1, str2, length=5):
+    """Check to strings word-by-word based for word length
+
+    The strings are checked word for word, but only for the first
+    `length` characters
+
+    Parameters
+    ----------
+    str1, str2: str
+        The strings to compare
+
+    length: int
+        The number of characters in each word to check.
+
+    Returns
+    -------
+    match: boolean
+        True if the strings match
+    """
+    words1 = str1.split()
+    words2 = str2.split()
+    if len(words1) != len(words2):
+        return False
+    for w1, w2 in zip(words1, words2):
+        if w1[:length] != w2[:length]:
```

Comment:
This is case-sensitive. I am not sure what you use this for, but if you are checking FITS keywords, perhaps you also want to add in an option for case-insensitive comparison?